src/csharp/Microsoft.Spark.E2ETest/SparkFixture.cs

```diff
@@ -114,6 +115,36 @@ public SparkFixture()
             Jvm = ((IJvmObjectReferenceProvider)Spark).Reference.Jvm;
         }
 
+        public string AddPackages(string args)
+        {
+            if (args != "")
+            {
+                List<string> argSplit = new(args.Split(' '));
+                int packageIndex = argSplit.FindIndex(a => a == "--packages");
+                string package = argSplit[packageIndex + 1];
+                argSplit[packageIndex] = "";
+                argSplit[packageIndex + 1] = "";
+                return $"{string.Join(" ", argSplit.ToArray())} --packages {package},{GetAvroPackage()}";
+            }
+            else
+            {
+                return $"--packages {GetAvroPackage()}";
+            }
```

Comment:
Looks like this changes the order of the args a bit, ie always puts --packages at the end even if it was defined at the beginning or middle, which isn't too bad.   But seems to also put in extra spaces, 

ie
"--packages abc --conf 1 --conf 2"  => "`  `--conf 1 --conf 2 --packages abc,avropackage"
"--conf 1 --conf 2 --packages abc --conf 3" => "--conf 1 --conf 2`  ` --conf 3 --packages abc,avropackage"

How about something like this instead, will this work?  wdyt ?

```csharp
        public string AddPackages(string args)
        {
            string packagesOption = "--packages ";
            string[] splits = args.Split(packagesOption, 2);
            
            StringBuilder newArgs = new StringBuilder(splits[0])
                .Append(packagesOption)
                .Append(GetAvroPackage());
            if (splits.Length > 1)
            {
                newArgs.Append(",").Append(splits[1]);
            }
            
            return newArgs.ToString();
        }
```

Reply:
Done!